openmetadata-ui/src/main/resources/ui/src/hooks/useAirflowStatus.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { AxiosError } from 'axios';
+import { useEffect, useState } from 'react';
+import { checkAirflowStatus } from '../axiosAPIs/ingestionPipelineAPI';
+
+interface UseAirflowStatusProps {
+  isFetchingStatus: boolean;
+  isAirflowAvailable: boolean;
+  error: AxiosError | undefined;
+  fetchAirflowStatus: () => Promise<void>;
+}
+
+export const useAirflowStatus = (): UseAirflowStatusProps => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isAirflowAvailable, setIsAirflowAvailable] = useState<boolean>(false);
+  const [error, setError] = useState<AxiosError>();
+
+  const fetchAirflowStatus = async () => {
+    setIsLoading(true);
+    try {
+      const response = await checkAirflowStatus();
+      setIsAirflowAvailable(response.status === 200);
+    } catch (error) {
+      setError(error as AxiosError);
+      setIsAirflowAvailable(false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchAirflowStatus();
```

Comment:
Does this means it only runs once? Or it will run per file load?
How can user apply for refetch status or they need to refresh only?

Reply:
The status will be fetched on the component mount and we are exposing the `fetchAirflowStatus` so we can fetch the status on demand.